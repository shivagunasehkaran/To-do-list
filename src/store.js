// redux specific imports
import {applyMiddleware, createStore, compose} from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers/rootReducer';

let middleware = applyMiddleware(thunk);

const store = createStore(rootReducer, compose(middleware));

export default store;
