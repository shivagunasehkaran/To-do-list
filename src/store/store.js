// redux specific imports
import AsyncStorage from '@react-native-async-storage/async-storage';
import {applyMiddleware, createStore} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';

// Middleware : Redux Persist Config
const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  //Save specific Reducers
  whitelist: ['todo'],
  // Don't save specific Reducers
  blacklist: [],
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);

export {store, persistor};
