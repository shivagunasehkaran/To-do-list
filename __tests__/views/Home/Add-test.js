import enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {mockStore} from '../../../src/store/__mocks__/store';
import ConnectedAdd from '../../../src/views/Add/Add';

enzyme.configure({adapter: new Adapter()});

const defaultStore = {
  network: {isConnected: true},
};

describe('ConnectedAdd screen: Snapshot Testing', () => {
  it('renders correctly when success callback with all the required params in response', async () => {
    const data = {
      title: 'shiva',
      description: 'test',
    };
    const store = mockStore({
      ...defaultStore,
      todo: {
        todoDetails: {
          title: 'test',
          description: 'test',
        },
      },
      todoData: {data: data},
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedAdd />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ConnectedAdd render values and components', () => {
  it('renders correctly with default', async () => {
    const store = mockStore({
      ...defaultStore,
      navigation: {navigate: jest.fn()},
      todo: {
        todoDetails: {
          title: 'test',
          description: 'test',
        },
      },
    });

    const navigateSpy = jest.mock('@react-navigation/native', () => {
      return {
        useNavigation: () => ({goBack: jest.fn()}),
        useRoute: jest.fn(),
      };
    });

    const saveTodoDetailsMock = jest.fn();

    const wrapper = shallow(
      <Provider store={store}>
        <ConnectedAdd store={store} saveTodoDetails={saveTodoDetailsMock} />
      </Provider>,
    );
    const screen = wrapper.find('Add');
    const actionScreen = screen
      .findWhere(node => node.prop('testID') === 'addAction')
      .at(0);
    actionScreen
      .findWhere(node => node.prop('testID') === 'addAction')
      .at(0)
      .simulate('click');
    expect(navigateSpy).toHaveBeenCalledTimes(1);
  });
});
