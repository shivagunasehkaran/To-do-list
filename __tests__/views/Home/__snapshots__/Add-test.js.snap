// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedAdd screen: Snapshot Testing renders correctly when success callback with all the required params in response 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "margin": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Avenir-Medium",
            "fontSize": 20,
            "margin": 15,
          }
        }
      >
        Title
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter Title"
        rejectResponderTermination={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#000000",
            "borderRadius": 5,
            "elevation": 1,
            "flexDirection": "row",
            "fontSize": 20,
            "height": 50,
            "paddingLeft": 15,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Avenir-Medium",
            "fontSize": 20,
            "margin": 15,
          }
        }
      >
        Description
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter Description"
        rejectResponderTermination={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#000000",
            "borderRadius": 5,
            "elevation": 1,
            "flexDirection": "row",
            "fontSize": 20,
            "height": 50,
            "paddingLeft": 15,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#8F8FFF",
          "bottom": 0,
          "height": 70,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": "100%",
        }
      }
      testID="addAction"
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Avenir-Medium",
            "fontSize": 20,
            "margin": 10,
          }
        }
      >
        SAVE
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
